dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/EnrollmentImportValidationTest.java

```diff
@@ -0,0 +1,680 @@
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.core.Every.everyItem;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Slf4j
+public class EnrollmentImportValidationTest
+    extends AbstractImportValidationTest
+
+{
+    @Autowired
+    protected TrackedEntityInstanceService trackedEntityInstanceService;
+
+    @Autowired
+    private TrackerBundleService trackerBundleService;
+
+    @Autowired
+    private ObjectBundleService objectBundleService;
+
+    @Autowired
+    private ObjectBundleValidationService objectBundleValidationService;
+
+    @Autowired
+    private DefaultTrackerValidationService trackerValidationService;
+
+    @Autowired
+    private RenderService _renderService;
+
+    @Autowired
+    private UserService _userService;
+
+    @Override
+    protected void setUpTest()
+        throws IOException
+    {
+        renderService = _renderService;
+        userService = _userService;
+
+        Map<Class<? extends IdentifiableObject>, List<IdentifiableObject>> metadata = renderService.fromMetadata(
+            new ClassPathResource( "tracker/tracker_basic_metadata.json" ).getInputStream(), RenderFormat.JSON );
+
+        ObjectBundleParams params = new ObjectBundleParams();
+        params.setObjectBundleMode( ObjectBundleMode.COMMIT );
+        params.setImportStrategy( ImportStrategy.CREATE );
+        params.setObjects( metadata );
+
+        ObjectBundle bundle = objectBundleService.create( params );
+        ObjectBundleValidationReport validationReport = objectBundleValidationService.validate( bundle );
+        List<ErrorReport> errorReports = validationReport.getErrorReports();
+        assertTrue( errorReports.isEmpty() );
+
+        ObjectBundleCommitReport commit = objectBundleService.commit( bundle );
+        List<ErrorReport> objectReport = commit.getErrorReports();
+        assertTrue( objectReport.isEmpty() );
+
+        TrackerBundleParams trackerBundleParams = createBundleFromJson(
+            "tracker/validations/enrollments_te_te-data.json" );
+
+        TrackerBundle trackerBundle = trackerBundleService.create( trackerBundleParams ).get( 0 );
+        assertEquals( 4, trackerBundle.getTrackedEntities().size() );
+
+        TrackerValidationReport report = trackerValidationService.validate( trackerBundle );
+        assertEquals( 0, report.getErrorReports().size() );
+
+        TrackerBundleReport bundleReport = trackerBundleService.commit( trackerBundle );
+        assertEquals( TrackerStatus.OK, bundleReport.getStatus() );
+    }
+
+    @Test
+    public void testEnrollmentValidationOkAll()
```

Comment:
This method shows duplication of code with `TrackedEntityImportValidatioTest.java`.